src/modules/gateway/gateway.usecase.spec.ts

```diff
@@ -592,5 +592,221 @@ describe('GatewayUseCases', () => {
+        jest
+          .spyOn(workspaceUseCases, 'findOne')
+          .mockResolvedValueOnce(workspace);
+
+        await expect(
+          service.updateWorkspace(owner.uuid, {
+            maxSpaceBytes,
+            numberOfSeats,
+          }),
+        ).rejects.toThrow(NotFoundException);
+      });
+
+      it('When tier ID is invalid, then it should throw BadRequestException', async () => {
+        const invalidTierId = v4();
+        jest
+          .spyOn(userRepository, 'findByUuid')
+          .mockResolvedValueOnce(owner)
+          .mockResolvedValueOnce(workspaceUser);
+        jest
+          .spyOn(workspaceUseCases, 'findOne')
+          .mockResolvedValueOnce(workspace);
+        jest.spyOn(featureLimitService, 'getTier').mockResolvedValueOnce(null);
+
+        await expect(
+          service.updateWorkspace(owner.uuid, { tierId: invalidTierId }),
+        ).rejects.toThrow(BadRequestException);
+
+        expect(featureLimitService.getTier).toHaveBeenCalledWith(invalidTierId);
+      });
+
+      it('When update fails, then it should propagate the error', async () => {
+        const tierId = v4();
+        const updateError = new Error('Failed to update tier');
+        jest
+          .spyOn(userRepository, 'findByUuid')
+          .mockResolvedValueOnce(owner)
+          .mockResolvedValueOnce(workspaceUser);
+        jest
+          .spyOn(workspaceUseCases, 'findOne')
+          .mockResolvedValueOnce(workspace);
+        jest
+          .spyOn(featureLimitService, 'getTier')
+          .mockResolvedValueOnce(validTier);
+        jest
+          .spyOn(userRepository, 'updateBy')
+          .mockRejectedValueOnce(updateError);
+
+        await expect(
+          service.updateWorkspace(owner.uuid, { tierId }),
+        ).rejects.toThrow(updateError);
+      });
+
+      describe('when updating tier', () => {
+        it('When tier exists and is different from workspace user tier, then it should validate and update tier', async () => {
+          const tier = newTier();
+          jest
+            .spyOn(userRepository, 'findByUuid')
+            .mockResolvedValueOnce(owner)
+            .mockResolvedValueOnce(workspaceUser);
+          jest
+            .spyOn(workspaceUseCases, 'findOne')
+            .mockResolvedValueOnce(workspace);
+          jest
+            .spyOn(featureLimitService, 'getTier')
+            .mockResolvedValueOnce(tier);
+          jest
+            .spyOn(userRepository, 'updateBy')
+            .mockResolvedValueOnce(undefined);
+
+          await service.updateWorkspace(owner.uuid, {
+            tierId: tier.id,
+          });
+
+          expect(featureLimitService.getTier).toHaveBeenCalledWith(tier.id);
+          expect(userRepository.updateBy).toHaveBeenCalledWith(
+            { uuid: workspaceUser.uuid },
+            { tierId: tier.id },
+          );
+        });
+
+        it('When tier exists and is same as workspace user tier, then it should not update tier', async () => {
```

Comment:
This test is missing the check to ensure the `updateBy` function has not been called